src/main/java/appeng/crafting/CraftingJob.java

```diff
@@ -375,22 +374,22 @@ private void logCraftingJob( String type, Stopwatch timer )
 			final long elapsedTime = timer.elapsed( TimeUnit.MILLISECONDS );
 			final String actionSource;
 
-			if( this.actionSrc instanceof MachineSource )
+			if( this.actionSrc.player().isPresent() )
 			{
-				final IActionHost machineSource = ( (MachineSource) this.actionSrc ).via;
+				final PlayerSource source = (PlayerSource) this.actionSrc;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "source". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "source" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1481)